tortoise/tortoise.go

```diff
@@ -129,6 +138,9 @@ func (t *turtle) evict(ctx context.Context) {
 
 // EncodeVotes by choosing base ballot and explicit votes.
 func (t *turtle) EncodeVotes(ctx context.Context, conf *encodeConf) (*types.Opinion, error) {
+	if err := t.checkDrained(); err != nil {
```

Comment:
for me this interface is more straightforward to either
- `checkDrained` return boolean indicating drained or not, and return the formatted error here
- or just drain the the list right here. as if we are creating ballot, the beacon is available.

Reply:
i wrote it this way as it seems easier to remove in future, and i also want to call tally votes when encode votes is called, but current code doesn't do it